whois/README.md

```diff
@@ -1,13 +1,120 @@
 # polkadot-whois
 
-Tracking of validators using on-chain and substrate telemetry data. The data can be queried by validator's `AccountId` or `(session_index, validator_index)`.
+A tool for converting and querying information about validators using interchangeable unique identifiers, information about a validator can be queried using the following indentifiers:
+- Validator index in `paraSessionInfo` pallet.
+- The account key.
+- The authority discovery id, provided in the hex format, it is the same format as what is displayed in `session.next_keys`.
+- The peer id, the network peer id used for connecting to other nodes.
 
-Usage:
+It prints a list of information for each queried validator, G.g:
+```
+validator_index=<INDEX<>>, account=<ACCOUNT>, peer_id=<PEER_ID>, authorithy_id_discover=0x<AUTHORITY_DISCOVERY>, addresses=<KNOWN_ADDRESSES>, version="POLKADOT_NODE_VERSION"
+```
+
+It uses the subp2p-explorer for querying the DHT information about each validator in the network.
+
+## Usage:
 
+### Query by validator index
+```
+cargo run  --bin polkadot-whois -- \
```

Comment:
It's a pretty big query. Can we reduce it a bit? I think we can take a genesis from the RPC.

Reply:
Yeah, can do that.

Reply:
Done.